### PR TITLE
Api derive

### PR DESCRIPTION
**Pull request by cinlk**
migrate api-derive from apps

### CODE REVIEW

File: packages/api-derive/src/staking/query.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  api: ApiInterfaceRx
+): (accountId: Uint8Array | string, queuedKeys: Vec<ITuple<[AccountId, Keys]>>) => Observable<DeriveStakingQuery> {
+  return memo(
+    instanceId,
+    (accountId: Uint8Array | string, queuedKeys: Vec<ITuple<[AccountId, Keys]>>): Observable<DeriveStakingQuery> => {
+      const stashId: AccountId = api.registry.createType('AccountId', accountId);
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      return (api.query.staking.erasStakers ? retrieveCurr(api, stashId) : retrievePrev(api, stashId)).pipe(
```
Comment (sxlwar) [inline packages/api-derive/src/staking/query.ts:114]: Help to confirm whether the `api.query.staking.erasStakers` method always exists? In that case, the logic here could be more streamlined. @WoeOm @AurevoirXavier 
image

File: packages/api-derive/src/staking/query.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  queuedKeys: Vec<ITuple<[AccountId, Keys]>>,
+  [controllerIdOpt, nominatorsOpt, rewardDestination, validatorPrefs, nextKeys, exposure]: MultiResult
+): Observable<DeriveStakingQuery> {
+  const controllerId = controllerIdOpt.unwrapOr(null);
+  // const nominators = nominatorsOpt.unwrapOr(null);
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  return controllerId
+    ? api.query.staking.ledger(controllerId).pipe(
```
Comment (sxlwar) [inline packages/api-derive/src/staking/query.ts:83]: The same as Line 117

File: packages/api-derive/src/staking/query.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
+// Copyright 2017-2022 @darwinia/api-derive authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
```
Comment (sxlwar) [inline packages/api-derive/src/staking/query.ts:4]: Need to evaluate these disabled eslint rules. Whether the errors suggested by these rules are real or not

Check other files as well

**Review by sxlwar — CHANGES_REQUESTED** [review]

File: packages/api-derive/src/staking/query.ts (hunk 3 above)
Comment (cinlk) [reply packages/api-derive/src/staking/query.ts:4]: > Need to evaluate these disabled eslint rules. Whether the errors suggested by these rules are real or not
> 
> Check other files as well

I  have try to remove those "eslint-disable" comments for entire file at each of api-derive project files,  and then executing the command "tsc --emitDeclarationOnly --outdir ./build" successfully compile files.

File: packages/api-derive/src/staking/query.ts (hunk 1 above)
Comment (AurevoirXavier) [reply packages/api-derive/src/staking/query.ts:114]: I'm don't know if they will change the API in the future.

But it still exists in Polkadot v0.9.18.

Comment (acgxv) [reply packages/api-derive/src/staking/query.ts:114]: I'm don't know if they will change the API in the future.

But it still exists in Polkadot v0.9.18.

File: packages/api-derive/src/staking/query.ts (hunk 2 above)
Comment (acgxv) [reply packages/api-derive/src/staking/query.ts:83]: Same.

**Review by WoeOm — APPROVED** [review]

**Pull request merged**